winch/codegen/src/visitor.rs

```diff
@@ -609,14 +612,97 @@ where
         }
     }
 
-    // TODO verify the case where the target local is on the stack.
+    // TODO: verify the case where the target local is on the stack.
     fn visit_local_set(&mut self, index: u32) {
         let src = self.context.set_local(self.masm, index);
         self.context.free_reg(src);
     }
 
     fn visit_call(&mut self, index: u32) {
-        self.emit_call(FuncIndex::from_u32(index));
+        let callee = self.env.callee_from_index(FuncIndex::from_u32(index));
+        self.emit_call(callee);
+    }
+
+    fn visit_call_indirect(&mut self, type_index: u32, table_index: u32, _: u8) {
+        let type_index = TypeIndex::from_u32(type_index);
+        let table_index = TableIndex::from_u32(table_index);
+        let table_data = self.env.resolve_table_data(table_index);
+        let ptr_type = self.env.ptr_type();
+
+        let builtin = self
+            .env
+            .builtins
+            .table_get_lazy_init_func_ref::<M::ABI, M::Ptr>();
+
+        FnCall::new_unchecked(&builtin.sig).with_lib(
```

Comment:
I'm not quite understanding the "unchecked" mechanism here, or the need to defer calculation of stack size needed, etc. -- don't we know the signatures for all our builtins? Could you add a comment describing why this needs to be unchecked?

Reply:
I've added more documentation in the function definition, hope this clarifies things.